pkg/pillar/types/downloadertypes.go

```diff
@@ -167,3 +168,71 @@ func AllowNonFreePort(gc ConfigItemValueMap, objType string) bool {
 		return false
 	}
 }
+
+// ResolveConfig key/index to this is the ImageID which is allocated by the controller.
+// It will resolve container image tag to sha256
+type ResolveConfig struct {
+	AppID            uuid.UUID
+	ImageID          uuid.UUID
+	DatastoreID      uuid.UUID
+	Name             string
+	NameIsURL        bool // If not we form URL based on datastore info
+	AllowNonFreePort bool
+	PurgeCounter     uint32
+}
+
+// Key : ImageID is used to differentiate different config
+func (config ResolveConfig) Key() string {
+	return fmt.Sprintf("%s-%s-%v", config.AppID.String(), config.ImageID.String(), config.PurgeCounter)
+}
+
+// VerifyFilename will verify the key name
+func (config ResolveConfig) VerifyFilename(fileName string) bool {
+	expect := config.Key() + ".json"
+	ret := expect == fileName
+	if !ret {
+		log.Errorf("Mismatch between filename and contained key: %s vs. %s\n",
+			fileName, expect)
+	}
+	return ret
+}
+
+// ResolveStatus : The key/index to this is the ImageID which comes from ResolveConfig.
+type ResolveStatus struct {
+	AppID        uuid.UUID
+	ImageID      uuid.UUID
```

Comment:
Ditto. But you need to add the datastore id in here so that the key can be the same for the Config and Status.

Reply:
Removed.